src/internal/middleware/logging/interceptor.go

```diff
@@ -0,0 +1,462 @@
+//nolint:wrapcheck
+package logging
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"reflect"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/pachyderm/pachyderm/v2/src/auth"
+	"github.com/pachyderm/pachyderm/v2/src/enterprise"
+	"github.com/pachyderm/pachyderm/v2/src/identity"
+	"github.com/pachyderm/pachyderm/v2/src/license"
+)
+
+type logConfig struct {
+	// interfaces here are the request and response protobufs, respectively
+	level             func(interface{}, interface{}, error) logrus.Level
```

Comment:
it looks like we aren't using the request/response arguments for logConfig.level(). What're your thoughts for having them in the signature?

Reply:
I'd be ok with removing it - I figured for customization it'd be nice to have all the information right there, but if it's ever needed (and it probably won't be), it's not too hard to add back in.